src/components/Requirements.vue

```diff
@@ -1,214 +1,573 @@
+        //       name: 'CALS Credits',
+        //       type: 'Credits',
+        //       fulfilled: 12,
+        //       required: 55,
+        //       display: false
+        //     },
+        //     {
+        //       name: 'PE Credits',
+        //       type: 'Credits',
+        //       fulfilled: 1,
+        //       required: 2,
+        //       display: false
+        //     }
+        //   ],
+        //   completed: [
+        //     {
+        //       name: 'Quantitative Literacy',
+        //       type: 'Credits',
+        //       fulfilled: 2,
+        //       required: 2,
+        //       display: false
+        //     }
+        //   ]
+        // }
       ]
     };
   },
 
   methods: {
-    turnDetails(index, bool) {
-      this.reqs[index].displayDetails = bool;
+    toggleDetails(index) {
+      this.reqs[index].displayDetails = !this.reqs[index].displayDetails;
     },
+
+    turnSubDetails(index, id, bool) {
+      this.reqs[index].ongoing[id].display = bool;
+    },
+
     turnCompleted(index, bool) {
       this.reqs[index].displayCompleted = bool;
+    },
+
+    getCourseCodesArray() {
+      const courses = [];
+      this.semesters.forEach(semester => {
+        semester.courses.forEach(course => {
+          courses.push(`${course.subject} ${course.code}`);
+        });
+      });
+
+      return courses;
+    },
+
+    async getReqs(coursesTaken, college, major) {
+      // TODO: make it so that it takes in classes corresponding with years/semesters for most accurate information
+      const coursesTakenWithInfo = {};
+      const courseData = await Promise.all(
+        coursesTaken.map(courseTaken => getCourseInfo(courseTaken))
+      );
+
+      for (let i = 0; i < coursesTaken.length; i += 1) { coursesTakenWithInfo[coursesTaken[i]] = courseData[i]; }
+
+      // prepare final output JSONs
+      const finalRequirementJSONs = [];
+
+      // PART 1: check university requirements
+      if (!reqsData.university) throw new Error('University requirements not found.');
+      const universityReqs = reqsData.university;
+      finalRequirementJSONs.push({
+        groupName: 'University',
+        reqs: await iterateThroughRequirements(coursesTakenWithInfo, universityReqs.requirements, 'university')
+      });
+
+      // PART 2: check college requirements
+      if (!(college in reqsData.college)) throw new Error('College not found.');
+      const collegeReqs = reqsData.college[college];
+      finalRequirementJSONs.push({
+        groupName: 'College',
+        reqs: await iterateThroughRequirements(coursesTakenWithInfo, collegeReqs.requirements, 'college')
+      });
+
+      // PART 3: check major reqs
+      if (!(major in reqsData.major)) throw new Error('Major not found.');
+      const majorReqs = reqsData.major[major];
+      finalRequirementJSONs.push({
+        groupName: 'Major',
+        reqs: await iterateThroughRequirements(coursesTakenWithInfo, majorReqs.requirements, 'major')
```

Comment:
Is it the best idea to have these 3 awaits in a row? Is there a better way to do this?

Reply:
Think it would approve the load time of the menu if they were all asynchronous. This is likely a temporary solution to load the data in order